javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,83 @@
+// Voice assistant
+
+let myName = '';
+const toDo = [];
+
+function getReply(command) {
+    const name = command.split(' ');
+    if (command.startsWith('Hello my name is')) {
+        if (myName === '') {
+            myName = name[name.length - 1];
+            return `Nice to meet you ${myName}!`;
+        } else {
+            return 'You wrote twice!';
+        }
+    } else if (command === 'What is my name') {
+        if (myName === '') {
+            return "You haven't introduced yourself yet!"
+        } else {
+            return `Your name is ${myName}!`;
+        }
+    } else if (command.startsWith('Add') && command.endsWith('to my todo')) {
+        let clear = command.replace('Add ', '');
+        let activity = clear.replace(' to my todo', '');
+        toDo.push(activity);
+        return `${activity} added to your todo!`;
+    } else if (command.startsWith('Remove') && command.endsWith('from my todo')) {
+        let clear = command.replace('Remove ', '');
+        let activity = clear.replace(' from my todo', '');
```

Comment:
I recommend that you start using higher order functions. It does the job with less lines of coed.